version: '3.1'

services:

  database:
    image: postgres:14.2
    restart: always
    container_name: debezium-cdc-rabbit-db
    networks:
      - docker-local-network
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: debezium-cdc-rabbit-db
    command: ['postgres', '-c', 'wal_level=logical']
    volumes:
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    ports:
      - 5442:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 3s
      retries: 5      

  rabbitmq:
    image: rabbitmq:3-management
    container_name: debezium-cdc-rabbit-broker
    networks:
      - docker-local-network 
    ports:
      - 5672:5672
      - 15672:15672
    volumes:
      - ./rabbitmq/definitions.json:/etc/rabbitmq/definitions.json
      - ./rabbitmq/rabbitmq.conf:/etc/rabbitmq/rabbitmq.conf
    healthcheck:
      test: ["CMD", "rabbitmq-diagnostics", "ping"]
      interval: 5s
      timeout: 3s
      retries: 5

  debezium:
    image: debezium/server:2.7.3.Final
    container_name: debezium
    ports:
      - "8080:8080"
    volumes:
      - ./debezium/conf:/debezium/conf
      - ./debezium/data:/debezium/data
    networks:
      - docker-local-network
    depends_on:
      database:
        condition: service_healthy
      rabbitmq:
        condition: service_healthy     

networks: 
  docker-local-network:
    driver: bridge  